pkgs/applications/version-management/git-and-tools/default.nix

```diff
@@ -40,6 +40,10 @@ let
 
   git-fame = callPackage ./git-fame {};
 
+  git-filter-repo = callPackage ./git-filter-repo {
+    inherit (python3Packages) buildPythonApplication fetchPypi;
```

Comment:
```suggestion
    inherit (python3Packages) buildPythonApplication fetchPypi setuptools_scm;
```

Reply:
to avoid 
```
$ nix-build ./nixpkgs/ -A gitAndTools.git-filter-repo
these derivations will be built:
  /nix/store/ppvxnwxx0i8qwcgym4g8vsw1bpak2xzy-git-filter-repo-2.24.0.drv
building '/nix/store/ppvxnwxx0i8qwcgym4g8vsw1bpak2xzy-git-filter-repo-2.24.0.drv'...
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/fy14pm0cjr3qdbc84kc6ykxwcq6qi6f4-git-filter-repo-2.24.0.tar.gz
source root is git-filter-repo-2.24.0
setting SOURCE_DATE_EPOCH to timestamp 1577393070 of file git-filter-repo-2.24.0/setup.cfg
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
WARNING: The directory '/homeless-shelter/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff7079790>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5cd0090>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5cd02d0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5cd0490>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5cd0550>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/
ERROR: Could not find a version that satisfies the requirement setuptools_scm (from versions: none)
ERROR: No matching distribution found for setuptools_scm
Traceback (most recent call last):
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/setuptools/installer.py", line 128, in fetch_build_egg
    subprocess.check_call(cmd)
  File "/nix/store/bzvpdbzdy3a1qba8f9ynrzyz811c10pp-python3-3.7.6/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/bzvpdbzdy3a1qba8f9ynrzyz811c10pp-python3-3.7.6/bin/python3.7', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmpe9jr51r_', '--quiet', 'setuptools_scm']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 5, in <module>
    setup(use_scm_version=dict(root="..", relative_to=__file__))
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/setuptools/__init__.py", line 144, in setup
    _install_setup_requires(attrs)
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/setuptools/__init__.py", line 139, in _install_setup_requires
    dist.fetch_build_eggs(dist.setup_requires)
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/setuptools/dist.py", line 721, in fetch_build_eggs
    replace_conflicting=True,
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 782, in resolve
    replace_conflicting=replace_conflicting
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1065, in best_match
    return self.obtain(req, installer)
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1077, in obtain
    return installer(requirement)
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/setuptools/dist.py", line 777, in fetch_build_egg
    return fetch_build_egg(self, req)
  File "/nix/store/p1dmakxx9x71m3zwhh2w0908hhg8jj7v-python3.7-setuptools-42.0.2/lib/python3.7/site-packages/setuptools/installer.py", line 130, in fetch_build_egg
    raise DistutilsError(str(e))
distutils.errors.DistutilsError: Command '['/nix/store/bzvpdbzdy3a1qba8f9ynrzyz811c10pp-python3-3.7.6/bin/python3.7', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmpe9jr51r_', '--quiet', 'setuptools_scm']' returned non-zero exit status 1.
builder for '/nix/store/ppvxnwxx0i8qwcgym4g8vsw1bpak2xzy-git-filter-repo-2.24.0.drv' failed with exit code 1
```